### ISSUE TITLE
Use a forked nodejs tree instead of a set of patches?

### ISSUE THREAD

**Issue by gautaz** [vercel/pkg-fetch#172]
Hello,

I am wondering what is the main motivation for using a patch set instead of forking the NodeJS repo and maintaining a history based on this patched fork.

AFAIK, using a forked tree would provide the following benefits:
- follow the applied modifications over time
- being able to build patched NodeJS binaries whatever the wanted version is as long as there is a patched version provided in the history (this could be done for instance by identifying special branches/tags `-pkg` in the history)
- ease patching new NodeJS revisions based on older patches
- ease the proposal of patches for new NodeJS releases or issue fixes for already patched releases through pull requests made by the community on this forked NodeJS tree

`pkg-fetch` could also propose an option for using a patched source tree forked from this common patched source tree in order to test before asking for a pull request.

Don't misunderstand my proposal (I'm not even sure I am the first to ask for it), `pkg` and `pkg-fetch` are great tools but I see many issues related to:
- the latest NodeJS release that is not yet supported
- the need to package a project with an older version of NodeJS while still using the latest `pkg` (if not the user has to dig in `pkg` history for the version related to his needs)
- the need to experiment patches for different architectures

And I feel that these tasks are currently cumbersome to achieve due to the usage of this fixed list of patches per `pkg-fetch` version.

Am I the only one feeling this way ?
Is this something that can be taken into consideration or at least be debated?

**Comment by robertsLando** [comment]
Keep nodejs sources up to date is actually the biggest problem of pkg, IMO the best alternative would be to convince nodejs developers to integrate this feature in nodejs. There is an issue that ask about this: https://github.com/nodejs/node/issues/26026

**Comment by gautaz** [comment]
@robertsLando I totally agree, I wasn't aware of this NodeJS issue, this would be the ideal course of actions to avoid maintaining this outside of the NodeJS source tree.

Do you want me to keep this issue opened to trace the link to the nodejs issue in the meantime or perhaps keep it in case the nodejs issue doesn't end up well?

**Comment by robertsLando** [comment]
It's ok for me

**Comment by robertsLando** [comment]
There is also this: https://chromium-review.googlesource.com/c/v8/v8/+/1462961

It's an open PR from @danielgindi
